rinex/src/context/mod.rs

```diff
@@ -24,561 +22,268 @@ use rinex_qc_traits::HtmlReport;
+            RinexType::ObservationData => Self::Observation,
+            RinexType::NavigationData => Self::BroadcastNavigation,
+            RinexType::MeteoData => Self::MeteoObservation,
+            RinexType::ClockData => Self::HighPrecisionClock,
+            RinexType::IonosphereMaps => Self::Ionex,
+            RinexType::AntennaData => Self::Antex,
+        }
     }
 }
 
-/// RnxContext is a structure dedicated to RINEX post processing workflows,
-/// like precise timing, positioning or atmosphere analysis.
-#[derive(Default, Debug, Clone)]
-pub struct RnxContext {
-    /// Optional OBS RINEX: provides GNSS signals, allows precise positioning.
-    pub obs: Option<ProvidedData<Rinex>>,
-    /// Optional NAV RINEX: allows precise positioning.
-    pub nav: Option<ProvidedData<Rinex>>,
-    /// Optional CLK RINEX
-    pub clk: Option<ProvidedData<Rinex>>,
-    /// Optional ATX RINEX
-    pub atx: Option<ProvidedData<Rinex>>,
-    /// Optional SP3 Orbits. Allows precise positioning.
-    pub sp3: Option<ProvidedData<SP3>>,
-    /// Optional Meteo file. Provides detailed modeling and meteo surveying.
-    pub meteo: Option<ProvidedData<Rinex>>,
-    /// Optional IONEX for accurate Ionosphere compensation.
-    pub ionex: Option<ProvidedData<Rinex>>,
+#[derive(Clone)]
+enum BlobData<'a> {
+    /// RINEX content
+    Rinex(&'a Rinex),
+    /// SP3 content
+    Sp3(&'a SP3),
 }
 
-impl RnxContext {
-    /// Build a RINEX post processing context from either a directory or a single file
-    pub fn new(path: &PathBuf) -> Result<Self, Error> {
-        if path.is_dir() {
-            /* recursive builder */
-            Self::from_directory(path)
-        } else {
-            /* load a single file */
-            Self::from_path(path)
-        }
-    }
-    /*
-     * Builds Self from a single file
-     */
-    fn from_path(path: &PathBuf) -> Result<Self, Error> {
-        let mut ctx = Self::default();
-        ctx.load(path)?;
-        Ok(ctx)
-    }
-    /*
-     * Builds Self by recursive browsing
-     */
-    fn from_directory(path: &PathBuf) -> Result<Self, Error> {
-        let mut ret = RnxContext::default();
-        let walkdir = WalkDir::new(path.to_string_lossy().to_string()).max_depth(5);
-        for entry in walkdir.into_iter().filter_map(|e| e.ok()) {
-            let path = entry.path().to_path_buf();
-            let filename = path
-                .file_name()
-                .ok_or(Error::FileNameDetermination)?
-                .to_string_lossy()
-                .to_string();
-
-            if !path.is_dir() {
-                match ret.load(&path) {
-                    Ok(_) => trace!("loaded \"{}\"", filename),
-                    Err(e) => error!("failed to load \"{}\", {:?}", path.display(), e),
-                }
-            }
+impl<'a> BlobData<'a> {
+    /// Returns reference to inner RINEX data.
+    pub fn as_rinex(&self) -> Option<&'a Rinex> {
+        match self {
+            Self::Rinex(r) => Some(r),
```

Comment:
@lnicola 

The hidden `BlobData` struct contains internal references to either RINEX or SP3.
Retrieving one of these references works perfectly